dashboard/app/models/pd/application/teacher1819_application.rb

```diff
@@ -430,5 +430,57 @@ def meets_criteria
     def total_score
       response_scores_hash.values.map {|x| x.try(:to_i)}.compact.reduce(:+)
     end
+
+    # Called once after the application is submitted, and the principal approval is done
+    def auto_score
+      responses = sanitize_form_data_hash
+
+      scores = {
+        regionalPartnerName: regional_partner ? YES : NO,
+        committed: responses[:committed] == YES ? YES : NO,
+        ableToAttendSingle: yes_no_response_to_yes_no_score(responses[:able_to_attend_single]),
+        principalApproval: yes_no_response_to_yes_no_score(responses[:principal_approval]),
+      }
+
+      if responses[:principal_approval] == YES
+        scores.merge!(
+          {
+            schedule_confirmed: yes_no_response_to_yes_no_score(responses[:schedule_confirmed]),
+            diversity_recruitment: yes_no_response_to_yes_no_score(responses[:diversity_recruitment]),
+            free_lunch_percent: responses[:free_lunch_percent].to_f >= 50 ? 5 : 0,
+            underrepresented_minority_percent:  responses[:underrepresented_minority_percent].to_f >= 50 ? 5 : 0,
+            wont_replace_existing_course: responses[:wont_replace_existing_course].try(:start_with?, NO) ? 5 : nil,
+          }
+        )
+      end
+
+      if course == 'csp'
+        scores[:csp_which_grades] = responses[:csp_which_grades].any? ? YES : NO
```

Comment:
Should these types of scores be boolean, and then the client can choose what to display? Why store "Yes"/"No"?

Reply:
I went back and forth on this. I went with string values instead of bools for scores so I wouldn't have to write extra logic for Yes/No questions versus scores with numeric values.